for reference, here are the ansible 2.7 errors on zuul on a fedora 31 vm:

```
=================================== FAILURES ===================================
________________________ test_cmdline_cmdline_override _________________________
[gw2] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

    def test_cmdline_cmdline_override():
        try:
            private_data_dir = tempfile.mkdtemp()
            play = [{'hosts': 'all', 'tasks': [{'debug': {'msg': random_string()}}]}]
    
            path = os.path.join(private_data_dir, 'project')
            os.makedirs(path)
    
            playbook = os.path.join(path, 'main.yaml')
            with open(playbook, 'w') as f:
                f.write(json.dumps(play))
            path = os.path.join(private_data_dir, 'inventory')
            os.makedirs(path)
    
            inventory = os.path.join(path, 'hosts')
            with open(inventory, 'w') as f:
                f.write('[all]\nlocalhost ansible_connection=local')
    
            # privateip: removed --hosts command line option from test beause it is
            # not a supported combination of cli options
            #cmdline('run', private_data_dir, '-p', playbook, '--hosts', 'all', '--cmdline', '-e foo=bar')
            cmdline('run', private_data_dir, '-p', playbook, '--cmdline', '-e foo=bar')
>           assert main() == 0
E           assert 2 == 0
E             -2
E             +0

test/integration/test___main__.py:241: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [all] *********************************************************************


TASK [Gathering Facts] *********************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

____________________________ test_cmdline_playbook _____________________________
[gw3] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

    def test_cmdline_playbook():
        try:
            private_data_dir = tempfile.mkdtemp()
            play = [{'hosts': 'all', 'tasks': [{'debug': {'msg': random_string()}}]}]
    
            path = os.path.join(private_data_dir, 'project')
            os.makedirs(path)
    
            playbook = os.path.join(path, 'main.yaml')
            with open(playbook, 'w') as f:
                f.write(json.dumps(play))
    
            path = os.path.join(private_data_dir, 'inventory')
            os.makedirs(path)
    
            inventory = os.path.join(path, 'hosts')
            with open(inventory, 'w') as f:
                f.write('[all]\nlocalhost ansible_connection=local')
    
            cmdline('run', private_data_dir, '-p', playbook, '--inventory', inventory)
    
>           assert main() == 0
E           assert 2 == 0
E             -2
E             +0

test/integration/test___main__.py:192: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [all] *********************************************************************

TASK [Gathering Facts] *********************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

______________ test_callback_plugin_saves_custom_stats[playbook0] ______________
[gw1] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

executor = <ansible_runner.runner.Runner object at 0x7f7af7afb4d0>
playbook = {'custom_set_stat.yml': '\n- name: custom set_stat calls should persist to the local disk so awx can save them\n  connection: local\n  hosts: all\n  tasks:\n    - set_stats:\n        data:\n          foo: "bar"\n'}

    @pytest.mark.parametrize('playbook', [
    {'custom_set_stat.yml': '''
    - name: custom set_stat calls should persist to the local disk so awx can save them
      connection: local
      hosts: all
      tasks:
        - set_stats:
            data:
              foo: "bar"
    '''},  # noqa
    ])
    def test_callback_plugin_saves_custom_stats(executor, playbook):
        executor.run()
        for event in executor.events:
            event_data = event.get('event_data', {})
            if 'artifact_data' in event_data:
>               assert event_data['artifact_data'] == {'foo': 'bar'}
E               AssertionError: assert {} == {'foo': 'bar'}
E                 Right contains more items:
E                 {'foo': 'bar'}
E                 Full diff:
E                 - {}
E                 + {'foo': 'bar'}

test/integration/test_display_callback.py:266: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [custom set_stat calls should persist to the local disk so awx can save them] ***


TASK [Gathering Facts] *********************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

________________ test_callback_plugin_task_args_leak[playbook0] ________________
[gw2] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

executor = <ansible_runner.runner.Runner object at 0x7f94f4e02990>
playbook = {'no_log_on_ok.yml': '\n- name: args should not be logged when no_log is set at the task or module level\n  connection...echo "PRIVATE"\n      no_log: true\n    - uri: url=https://example.org url_username="PUBLIC" url_password="PRIVATE"\n'}

    @pytest.mark.parametrize('playbook', [
    {'no_log_on_ok.yml': '''
    - name: args should not be logged when no_log is set at the task or module level
      connection: local
      hosts: all
      gather_facts: no
      tasks:
        - shell: echo "PUBLIC"
        - shell: echo "PRIVATE"
          no_log: true
        - uri: url=https://example.org url_username="PUBLIC" url_password="PRIVATE"
    '''},  # noqa
    ])
    def test_callback_plugin_task_args_leak(executor, playbook):
        executor.run()
        events = list(executor.events)
        assert events[0]['event'] == 'playbook_on_start'
        assert events[1]['event'] == 'playbook_on_play_start'
    
        # task 1
        assert events[2]['event'] == 'playbook_on_task_start'
>       assert events[3]['event'] == 'runner_on_start'
E       AssertionError: assert 'runner_on_failed' == 'runner_on_start'
E         - runner_on_failed
E         + runner_on_start

test/integration/test_display_callback.py:187: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [args should not be logged when no_log is set at the task or module level] ***


TASK [shell] *******************************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

_________ test_callback_plugin_censoring_does_not_overwrite[playbook0] _________
[gw3] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

executor = <ansible_runner.runner.Runner object at 0x7fb676843d90>
playbook = {'loop_with_no_log.yml': '\n- name: playbook variable should not be overwritten when using no log\n  connection: local...ms:\n        - "echo helloworld!"\n    - debug: msg="{{ command_register.results|map(attribute=\'stdout\')|list }}"\n'}

    @pytest.mark.parametrize('playbook', [
    {'loop_with_no_log.yml': '''
    - name: playbook variable should not be overwritten when using no log
      connection: local
      hosts: all
      gather_facts: no
      tasks:
        - command: "{{ item }}"
          register: command_register
          no_log: True
          with_items:
            - "echo helloworld!"
        - debug: msg="{{ command_register.results|map(attribute='stdout')|list }}"
    '''},  # noqa
    ])
    def test_callback_plugin_censoring_does_not_overwrite(executor, playbook):
        executor.run()
        events = list(executor.events)
        assert events[0]['event'] == 'playbook_on_start'
        assert events[1]['event'] == 'playbook_on_play_start'
    
        # task 1
        assert events[2]['event'] == 'playbook_on_task_start'
        # Ordering of task and item events may differ randomly
>       assert set(['runner_on_start', 'runner_item_on_ok', 'runner_on_ok']) == set([data['event'] for data in events[3:6]])
E       AssertionError: assert {'runner_item...ner_on_start'} == {'playbook_on_...er_on_failed'}
E         Extra items in the left set:
E         'runner_on_start'
E         'runner_item_on_ok'
E         'runner_on_ok'
E         Extra items in the right set:
E         'playbook_on_stats'
E         'runner_item_on_failed'...
E         
E         ...Full output truncated (5 lines hidden), use '-vv' to show

test/integration/test_display_callback.py:225: AssertionError
----------------------------- Captured stdout call -----------------------------

PLAY [playbook variable should not be overwritten when using no log] ***********


TASK [command] *****************************************************************

failed: [localhost] (item=None) => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}

fatal: [localhost]: FAILED! => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

____________ test_callback_plugin_records_notify_events[playbook0] _____________
[gw1] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

executor = <ansible_runner.runner.Runner object at 0x7f7af7be2ad0>
playbook = {'handle_playbook_on_notify.yml': '\n- name: handle playbook_on_notify events properly\n  connection: local\n  hosts: ...g="My Handler"\n  tasks:\n    - debug: msg="My Task"\n      changed_when: true\n      notify:\n        - my_handler\n'}

    @pytest.mark.parametrize('playbook', [
    {'handle_playbook_on_notify.yml': '''
    - name: handle playbook_on_notify events properly
      connection: local
      hosts: all
      handlers:
        - name: my_handler
          debug: msg="My Handler"
      tasks:
        - debug: msg="My Task"
          changed_when: true
          notify:
            - my_handler
    '''},  # noqa
    ])
    @pytest.mark.skipif(ANSIBLE_VERSION < '2.5', reason="v2_playbook_on_notify doesn't work before ansible 2.5")
    def test_callback_plugin_records_notify_events(executor, playbook):
        executor.run()
        assert len(list(executor.events))
        notify_events = [x for x in executor.events if x['event'] == 'playbook_on_notify']
>       assert len(notify_events) == 1
E       assert 0 == 1
E         -0
E         +1

test/integration/test_display_callback.py:292: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [handle playbook_on_notify events properly] *******************************

TASK [Gathering Facts] *********************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

_____________________ test_module_level_no_log[playbook0] ______________________
[gw1] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

executor = <ansible_runner.runner.Runner object at 0x7f7af78eb6d0>
playbook = {'no_log_module_with_var.yml': '\n- name: ensure that module-level secrets are redacted\n  connection: local\n  hosts:...  url: https://example.org\n        url_username: john-jacob-jingleheimer-schmidt\n        url_password: "{{ pw }}"\n'}

    @pytest.mark.parametrize('playbook', [
    {'no_log_module_with_var.yml': '''
    - name: ensure that module-level secrets are redacted
      connection: local
      hosts: all
      vars:
        - pw: SENSITIVE
      tasks:
        - uri:
            url: https://example.org
            url_username: john-jacob-jingleheimer-schmidt
            url_password: "{{ pw }}"
    '''},  # noqa
    ])
    def test_module_level_no_log(executor, playbook):
        # It's possible for `no_log=True` to be defined at the _module_ level,
        # e.g., for the URI module password parameter
        # This test ensures that we properly redact those
        executor.run()
        assert len(list(executor.events))
>       assert 'john-jacob-jingleheimer-schmidt' in json.dumps(list(executor.events))
E       assert 'john-jacob-jingleheimer-schmidt' in '[{"uuid": "123bf5da-54f3-4b65-8e81-e9b9579e95d9", "counter": 1, "stdout": "", "start_line": 0, "end_line": 0, "runner... {"localhost": 1}, "rescued": 0, "skipped": {}, "artifact_data": {}, "uuid": "030b8e35-2894-4ea5-bc7b-49574c296a2c"}}]'
E        +  where '[{"uuid": "123bf5da-54f3-4b65-8e81-e9b9579e95d9", "counter": 1, "stdout": "", "start_line": 0, "end_line": 0, "runner... {"localhost": 1}, "rescued": 0, "skipped": {}, "artifact_data": {}, "uuid": "030b8e35-2894-4ea5-bc7b-49574c296a2c"}}]' = <function dumps at 0x7f7afc5a5ef0>([{'counter': 1, 'created': '2020-07-31T21:00:51.576003', 'end_line': 0, 'event': 'playbook_on_start', ...}, {'counter'...iled', ...}, {'counter': 5, 'created': '2020-07-31T21:00:55.650446', 'end_line': 9, 'event': 'playbook_on_stats', ...}])
E        +    where <function dumps at 0x7f7afc5a5ef0> = json.dumps
E        +    and   [{'counter': 1, 'created': '2020-07-31T21:00:51.576003', 'end_line': 0, 'event': 'playbook_on_start', ...}, {'counter'...iled', ...}, {'counter': 5, 'created': '2020-07-31T21:00:55.650446', 'end_line': 9, 'event': 'playbook_on_stats', ...}] = list(<generator object Runner.events at 0x7f7af7b8d9d0>)
E        +      where <generator object Runner.events at 0x7f7af7b8d9d0> = <ansible_runner.runner.Runner object at 0x7f7af78eb6d0>.events

test/integration/test_display_callback.py:318: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [ensure that module-level secrets are redacted] ***************************

TASK [Gathering Facts] *********************************************************
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

________________________________ test_role_run _________________________________
[gw2] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

    def test_role_run():
        rc = main(['-r', 'benthomasson.hello_role',
                   '--hosts', 'localhost',
                   '--roles-path', 'test/integration/roles',
                   'run',
                   "test/integration"])
>       assert rc == 0
E       assert 2 == 0
E         -2
E         +0

test/integration/test_main.py:123: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [localhost] ***************************************************************


TASK [Gathering Facts] *********************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

______________________________ test_role_logfile _______________________________
[gw2] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

    def test_role_logfile():
        try:
            rc = main(['-r', 'benthomasson.hello_role',
                       '--hosts', 'localhost',
                       '--roles-path', 'test/integration/project/roles',
                       '--logfile', 'new_logfile',
                       'run',
                       'test/integration'])
            assert os.path.exists('new_logfile')
>           assert rc == 0
E           assert 2 == 0
E             -2
E             +0

test/integration/test_main.py:146: AssertionError
----------------------------- Captured stdout call -----------------------------


PLAY [localhost] ***************************************************************


TASK [Gathering Facts] *********************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

_________________________ test_role_run_artifacts_dir __________________________
[gw2] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

    def test_role_run_artifacts_dir():
        rc = main(['-r', 'benthomasson.hello_role',
                   '--hosts', 'localhost',
                   '--roles-path', 'test/integration/roles',
                   '--artifact-dir', 'otherartifacts',
                   'run',
                   "test/integration"])
>       assert rc == 0
E       assert 2 == 0
E         -2
E         +0

test/integration/test_main.py:213: AssertionError
----------------------------- Captured stdout call -----------------------------

PLAY [localhost] ***************************************************************


TASK [Gathering Facts] *********************************************************

fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

___________________________ test_role_run_inventory ____________________________
[gw2] linux -- Python 3.7.7 /home/zuul/src/github.com/ansible/ansible-runner/.tox/ansible27/bin/python3

    def test_role_run_inventory():
    
        with temp_directory() as temp_dir:
            ensure_directory(os.path.join(temp_dir, 'inventory'))
            shutil.copy(os.path.join(HERE, 'inventory/localhost'), os.path.join(temp_dir, 'inventory/localhost'))
    
            rc = main(['-r', 'benthomasson.hello_role',
                       '--hosts', 'localhost',
                       '--roles-path', os.path.join(HERE, 'project/roles'),
                       '--inventory', 'localhost',
                       'run',
                       temp_dir])
>       assert rc == 0
E       assert 2 == 0
E         -2
E         +0

test/integration/test_main.py:279: AssertionError
----------------------------- Captured stdout call -----------------------------
/tmp/tmp7pm2s1pr

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: /usr/bin/python: No such file or directory\n", "module_stdout": "", "msg": "The module failed to execute correctly, you probably need to set the interpreter.\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   


PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   

=============================== warnings summary ===============================
```

--- a/test/conftest.py
+++ b/test/conftest.py
@@ -9,10 +9,16 @@ def mock_env_user(monkeypatch):
 
 
 @pytest.fixture(scope='session')
-def skipif_pre_ansible28():
+def is_pre_ansible28():
     try:
         if LooseVersion(pkg_resources.get_distribution('ansible').version) < LooseVersion('2.8'):
-            pytest.skip("Valid only on Ansible 2.8+")
+            return True
     except pkg_resources.DistributionNotFound:
         # ansible-base (e.g. ansible 2.10 and beyond) is not accessible in this way
         pass
+
+
+@pytest.fixture(scope='session')
+def skipif_pre_ansible28(is_pre_ansible28):
+    if is_pre_ansible28:
+        pytest.skip("Valid only on Ansible 2.8+")

--- a/test/conftest.py
+++ b/test/conftest.py
@@ -1,6 +1,18 @@
 import pytest
+from distutils.version import LooseVersion
+import pkg_resources
 
 
 @pytest.fixture(autouse=True)
 def mock_env_user(monkeypatch):
     monkeypatch.setenv("ANSIBLE_DEVEL_WARNING", "False")
+
+
+@pytest.fixture(scope='session')
+def skipif_pre_ansible28():
+    try:
+        if LooseVersion(pkg_resources.get_distribution('ansible').version) < LooseVersion('2.8'):
+            pytest.skip("Valid only on Ansible 2.8+")
+    except pkg_resources.DistributionNotFound:
+        # ansible-base (e.g. ansible 2.10 and beyond) is not accessible in this way
+        pass

--- a/test/integration/inventory/localhost_preansible28
+++ b/test/integration/inventory/localhost_preansible28
@@ -1,0 +1,2 @@
+[all]
+localhost ansible_connection=local ansible_python_interpreter-"/usr/bin/env python"

--- a/test/integration/test___main__.py
+++ b/test/integration/test___main__.py
@@ -168,7 +168,7 @@ def test_cmdline_role_with_playbook_option():
         assert exc == 1
 
 
-def test_cmdline_playbook():
+def test_cmdline_playbook(is_pre_ansible28):
     try:
         private_data_dir = tempfile.mkdtemp()
         play = [{'hosts': 'all', 'tasks': [{'debug': {'msg': random_string()}}]}]
@@ -185,7 +185,10 @@ def test_cmdline_playbook():
 
         inventory = os.path.join(path, 'hosts')
         with open(inventory, 'w') as f:
-            f.write('[all]\nlocalhost ansible_connection=local')
+            if is_pre_ansible28:
+                f.write('[all]\nlocalhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"')
+            else:
+                f.write('[all]\nlocalhost ansible_connection=local')
 
         cmdline('run', private_data_dir, '-p', playbook, '--inventory', inventory)
 
@@ -216,7 +219,7 @@ def test_cmdline_includes_one_option():
         assert exc == 1
 
 
-def test_cmdline_cmdline_override(skipif_pre_ansible28):
+def test_cmdline_cmdline_override(is_pre_ansible28):
     try:
         private_data_dir = tempfile.mkdtemp()
         play = [{'hosts': 'all', 'tasks': [{'debug': {'msg': random_string()}}]}]
@@ -232,7 +235,10 @@ def test_cmdline_cmdline_override(skipif_pre_ansible28):
 
         inventory = os.path.join(path, 'hosts')
         with open(inventory, 'w') as f:
-            f.write('[all]\nlocalhost ansible_connection=local')
+            if is_pre_ansible28:
+                f.write('[all]\nlocalhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"')
+            else:
+                f.write('[all]\nlocalhost ansible_connection=local')
 
         # privateip: removed --hosts command line option from test beause it is
         # not a supported combination of cli options

--- a/test/integration/test___main__.py
+++ b/test/integration/test___main__.py
@@ -216,7 +216,7 @@ def test_cmdline_includes_one_option():
         assert exc == 1
 
 
-def test_cmdline_cmdline_override():
+def test_cmdline_cmdline_override(skipif_pre_ansible28):
     try:
         private_data_dir = tempfile.mkdtemp()
         play = [{'hosts': 'all', 'tasks': [{'debug': {'msg': random_string()}}]}]

--- a/test/integration/test_display_callback.py
+++ b/test/integration/test_display_callback.py
@@ -176,7 +176,7 @@ def test_callback_plugin_no_log_filters(executor, playbook):
     - uri: url=https://example.org url_username="PUBLIC" url_password="PRIVATE"
 '''},  # noqa
 ])
-def test_callback_plugin_task_args_leak(executor, playbook):
+def test_callback_plugin_task_args_leak(executor, playbook, skipif_pre_ansible28):
     executor.run()
     events = list(executor.events)
     assert events[0]['event'] == 'playbook_on_start'
@@ -213,7 +213,7 @@ def test_callback_plugin_task_args_leak(executor, playbook):
     - debug: msg="{{ command_register.results|map(attribute='stdout')|list }}"
 '''},  # noqa
 ])
-def test_callback_plugin_censoring_does_not_overwrite(executor, playbook):
+def test_callback_plugin_censoring_does_not_overwrite(executor, playbook, skipif_pre_ansible28):
     executor.run()
     events = list(executor.events)
     assert events[0]['event'] == 'playbook_on_start'
@@ -258,7 +258,7 @@ def test_callback_plugin_strips_task_environ_variables(executor, playbook):
           foo: "bar"
 '''},  # noqa
 ])
-def test_callback_plugin_saves_custom_stats(executor, playbook):
+def test_callback_plugin_saves_custom_stats(executor, playbook, skipif_pre_ansible28):
     executor.run()
     for event in executor.events:
         event_data = event.get('event_data', {})
@@ -285,7 +285,7 @@ def test_callback_plugin_saves_custom_stats(executor, playbook):
 '''},  # noqa
 ])
 @pytest.mark.skipif(ANSIBLE_VERSION < '2.5', reason="v2_playbook_on_notify doesn't work before ansible 2.5")
-def test_callback_plugin_records_notify_events(executor, playbook):
+def test_callback_plugin_records_notify_events(executor, playbook, skipif_pre_ansible28):
     executor.run()
     assert len(list(executor.events))
     notify_events = [x for x in executor.events if x['event'] == 'playbook_on_notify']
@@ -309,7 +309,7 @@ def test_callback_plugin_records_notify_events(executor, playbook):
         url_password: "{{ pw }}"
 '''},  # noqa
 ])
-def test_module_level_no_log(executor, playbook):
+def test_module_level_no_log(executor, playbook, skipif_pre_ansible28):
     # It's possible for `no_log=True` to be defined at the _module_ level,
     # e.g., for the URI module password parameter
     # This test ensures that we properly redact those

--- a/test/integration/test_display_callback.py
+++ b/test/integration/test_display_callback.py
@@ -15,7 +15,7 @@ HERE = os.path.abspath(os.path.dirname(__file__))
 
 
 @pytest.fixture()
-def executor(tmpdir, request):
+def executor(tmpdir, request, is_pre_ansible28):
     private_data_dir = six.text_type(tmpdir.mkdir('foo'))
 
     playbooks = request.node.callspec.params.get('playbook')
@@ -28,9 +28,15 @@ def executor(tmpdir, request):
     # python interpreter used is not of much interest, we really want to silence warnings
     envvars['ANSIBLE_PYTHON_INTERPRETER'] = 'auto_silent'
 
+    if is_pre_ansible28:
+        inventory = 'localhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"'
+    else:
+        inventory = 'localhost ansible_connection=local'
+
+
     r = init_runner(
         private_data_dir=private_data_dir,
-        inventory="localhost ansible_connection=local",
+        inventory=inventory,
         envvars=envvars,
         playbook=yaml.safe_load(playbook)
     )

--- a/test/integration/test_events.py
+++ b/test/integration/test_events.py
@@ -10,12 +10,16 @@ from ansible_runner import run, run_async
 
 @pytest.mark.serial
 @pytest.mark.parametrize('containerized', [True, False])
-def test_basic_events(containerized, container_runtime_available, is_run_async=False,g_facts=False):
+def test_basic_events(containerized, container_runtime_available, is_pre_ansible28, is_run_async=False, g_facts=False):
     if containerized and not container_runtime_available:
         pytest.skip('container runtime(s) not available')
     tdir = tempfile.mkdtemp()
 
-    inventory = "localhost ansible_connection=local"
+    if is_pre_ansible28:
+        inventory = 'localhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"'
+    else:
+        inventory = 'localhost ansible_connection=local'
+
     playbook = [{'hosts': 'all', 'gather_facts': g_facts, 'tasks': [{'debug': {'msg': "test"}}]}]
     run_args = {'private_data_dir': tdir,
                 'inventory': inventory,
@@ -58,32 +62,44 @@ def test_basic_events(containerized, container_runtime_available, is_run_async=F
 
 @pytest.mark.serial
 @pytest.mark.parametrize('containerized', [True, False])
-def test_async_events(containerized, container_runtime_available):
-    test_basic_events(containerized, container_runtime_available, is_run_async=True,g_facts=True)
+def test_async_events(containerized, container_runtime_available, is_pre_ansible28):
+    test_basic_events(containerized, container_runtime_available, is_pre_ansible28, is_run_async=True,g_facts=True)
 
 
-def test_basic_serializeable():
+def test_basic_serializeable(is_pre_ansible28):
     tdir = tempfile.mkdtemp()
+    if is_pre_ansible28:
+        inv = 'localhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"'
+    else:
+        inv = 'localhost ansible_connection=local'
     r = run(private_data_dir=tdir,
-            inventory="localhost ansible_connection=local",
+            inventory=inv,
             playbook=[{'hosts': 'all', 'gather_facts': False, 'tasks': [{'debug': {'msg': "test"}}]}])
     events = [x for x in r.events]
     json.dumps(events)
 
 
-def test_event_omission():
+def test_event_omission(is_pre_ansible28):
     tdir = tempfile.mkdtemp()
+    if is_pre_ansible28:
+        inv = 'localhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"'
+    else:
+        inv = 'localhost ansible_connection=local'
     r = run(private_data_dir=tdir,
-            inventory="localhost ansible_connection=local",
+            inventory=inv,
             omit_event_data=True,
             playbook=[{'hosts': 'all', 'gather_facts': False, 'tasks': [{'debug': {'msg': "test"}}]}])
     assert not any([x['event_data'] for x in r.events])
 
 
-def test_event_omission_except_failed():
+def test_event_omission_except_failed(is_pre_ansible28):
     tdir = tempfile.mkdtemp()
+    if is_pre_ansible28:
+        inv = 'localhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"'
+    else:
+        inv = 'localhost ansible_connection=local'
     r = run(private_data_dir=tdir,
-            inventory="localhost ansible_connection=local",
+            inventory=inv,
             only_failed_event_data=True,
             playbook=[{'hosts': 'all', 'gather_facts': False, 'tasks': [{'fail': {'msg': "test"}}]}])
     all_event_datas = [x['event_data'] for x in r.events if x['event_data']]
@@ -100,10 +116,14 @@ def test_runner_on_start(rc, skipif_pre_ansible28):
     assert len(start_events) == 1
 
 
-def test_playbook_on_stats_summary_fields(rc):
+def test_playbook_on_stats_summary_fields(rc, is_pre_ansible28):
+    if is_pre_ansible28:
+        inv = 'localhost ansible_connection=local ansible_python_interpreter="/usr/bin/env python"'
+    else:
+        inv = 'localhost ansible_connection=local'
     tdir = tempfile.mkdtemp()
     r = run(private_data_dir=tdir,
-            inventory="localhost ansible_connection=local",
+            inventory=inv,
             playbook=[{'hosts': 'all', 'gather_facts': False, 'tasks': [{'debug': {'msg': "test"}}]}])
     stats_events = [x for x in filter(lambda x: 'event' in x and x['event'] == 'playbook_on_stats',
                                       r.events)]

--- a/test/integration/test_events.py
+++ b/test/integration/test_events.py
@@ -1,23 +1,11 @@
 import pytest
 import tempfile
-from distutils.version import LooseVersion
 from distutils.spawn import find_executable
-import pkg_resources
 import json
 import os
 import shutil
 
 from ansible_runner import run, run_async
-
-
-@pytest.fixture(scope='session')
-def skipif_pre_ansible28():
-    try:
-        if LooseVersion(pkg_resources.get_distribution('ansible').version) < LooseVersion('2.8'):
-            pytest.skip("Valid only on Ansible 2.8+")
-    except pkg_resources.DistributionNotFound:
-        # ansible-base (e.g. ansible 2.10 and beyond) is not accessible in this way
-        pass
 
 
 @pytest.mark.serial

--- a/test/integration/test_main.py
+++ b/test/integration/test_main.py
@@ -134,7 +134,7 @@ def test_role_run_abs():
     assert rc == 0
 
 
-def test_role_logfile():
+def test_role_logfile(skipif_pre_ansible28):
     try:
         rc = main(['-r', 'benthomasson.hello_role',
                    '--hosts', 'localhost',
@@ -279,7 +279,7 @@ def test_role_run_inventory():
     assert rc == 0
 
 
-def test_role_run_inventory_missing():
+def test_role_run_inventory_missing(skipif_pre_ansible28):
 
     with temp_directory() as temp_dir:
         ensure_directory(os.path.join(temp_dir, 'inventory'))

--- a/test/integration/test_main.py
+++ b/test/integration/test_main.py
@@ -114,7 +114,7 @@ def test_module_run_clean():
     assert rc == 0
 
 
-def test_role_run():
+def test_role_run(skipif_pre_ansible28):
     rc = main(['-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
                '--roles-path', 'test/integration/roles',
@@ -182,7 +182,7 @@ def test_role_bad_project_dir():
 
 
 @pytest.mark.serial
-def test_role_run_clean():
+def test_role_run_clean(skipif_pre_ansible28):
 
     rc = main(['-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
@@ -203,7 +203,7 @@ def test_role_run_cmd_line_abs():
     assert rc == 0
 
 
-def test_role_run_artifacts_dir():
+def test_role_run_artifacts_dir(skipif_pre_ansible28):
     rc = main(['-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
                '--roles-path', 'test/integration/roles',
@@ -214,7 +214,7 @@ def test_role_run_artifacts_dir():
     ensure_removed("test/integration/artifacts")
 
 
-def test_role_run_artifacts_dir_abs():
+def test_role_run_artifacts_dir_abs(skipif_pre_ansible28):
     try:
         with temp_directory() as temp_dir:
             rc = main(['-r', 'benthomasson.hello_role',
@@ -264,26 +264,28 @@ def test_role_run_args():
     assert rc == 0
 
 
-def test_role_run_inventory():
+def test_role_run_inventory(is_pre_ansible28):
 
+    inv = 'inventory/localhost_preansible28' if is_pre_ansible28 else 'inventory/localhost'
     with temp_directory() as temp_dir:
         ensure_directory(os.path.join(temp_dir, 'inventory'))
-        shutil.copy(os.path.join(HERE, 'inventory/localhost'), os.path.join(temp_dir, 'inventory/localhost'))
+        shutil.copy(os.path.join(HERE, inv), os.path.join(temp_dir, 'inventory/localhost'))
 
         rc = main(['-r', 'benthomasson.hello_role',
                    '--hosts', 'localhost',
                    '--roles-path', os.path.join(HERE, 'project/roles'),
-                   '--inventory', 'localhost',
+                   '--inventory', os.path.join(temp_dir, 'inventory/localhost'),
                    'run',
                    temp_dir])
     assert rc == 0
 
 
-def test_role_run_inventory_missing(skipif_pre_ansible28):
+def test_role_run_inventory_missing(is_pre_ansible28):
 
+    inv = 'inventory/localhost_preansible28' if is_pre_ansible28 else 'inventory/localhost'
     with temp_directory() as temp_dir:
         ensure_directory(os.path.join(temp_dir, 'inventory'))
-        shutil.copy(os.path.join(HERE, 'inventory/localhost'), os.path.join(temp_dir, 'inventory/localhost'))
+        shutil.copy(os.path.join(HERE, inv), os.path.join(temp_dir, 'inventory/localhost'))
 
         with pytest.raises(AnsibleRunnerException):
             main(['-r', 'benthomasson.hello_role',
@@ -309,14 +311,15 @@ def test_role_start():
 
 
 @pytest.mark.serial
-def test_playbook_start():
+def test_playbook_start(skipif_pre_ansible28):
 
+    inv = 'inventory/localhost'
     with temp_directory() as temp_dir:
         project_dir = os.path.join(temp_dir, 'project')
         ensure_directory(project_dir)
         shutil.copy(os.path.join(HERE, 'project/hello.yml'), project_dir)
         ensure_directory(os.path.join(temp_dir, 'inventory'))
-        shutil.copy(os.path.join(HERE, 'inventory/localhost'), os.path.join(temp_dir, 'inventory/localhost'))
+        shutil.copy(os.path.join(HERE, inv), os.path.join(temp_dir,'inventory/localhost'))
 
         # privateip: removed --hosts command line option from test beause it is
         # not a supported combination of cli options

